torchx/schedulers/aws_batch_scheduler.py

```diff
@@ -244,6 +245,10 @@ def _role_to_node_properties(
         "mountPoints": mount_points,
         "volumes": volumes,
     }
+    if role.num_replicas > 1:
+        instance_type = role.resource.capabilities.get(K8S_ITYPE, None)
```

Comment:
shouldn't the instance type get assigned on the server-side from the resource requirements and the type of instances in the AWS Batch CE?

Reply:
These are the requirements that restrict AWS Batch to only use the specified instance type (CE can have many instance types). For multinode though it's a must for instance type to be the same for all nodes (see the link I attached in the feature request)

Reply:
E.g. if you don't specify it and your CE only operates a single instance type - it will work, but only by chance.

Reply:
I see, is this specifically to launch between `p4d` and `p4de`? AFAIK there aren't any other accelerated instances that have the same resource specs (cpu, gpu, memMB), so the resource specs should be enough to disambiguate between instance types no?

Couple of comments:

1. We should document in `torchx.specs.named_resources_aws:K8S_TYPE` that this is also used for setting AWS Batch container instance type for non-homogeneous CEs and link issue #780 for details.
2. For custom named resources (see [documentation](https://pytorch.org/torchx/latest/advanced.html#registering-named-resources)) this won't work unless the user remembers to add `K8S_ITYPE` to `Resource.capabilities`  of the custom named resource they are adding. We should update the docs or better yet display a warning (`warnings.warn`) when the (CE is non-homogeneous &&  job is MNP  && resource.capabilities is missing K8S_TYPE) that one should be added to the custom named resource.

Reply:
Right, @kiukchung `p4d` and `p4de` are the two examples of _public_ resource types (e.g. training job requiring precisely `p4de` to get 80GB of GPU memory and if it gets a `p4d` it will OOM).

But it's also applicable to custom resources.

I was looking for a better mechanism to extract a scheduler-specific "metadata". Creating a type hierarchy is perhaps is an overkill, e.g. :
```
class AwsResource(Resource)
    instance_type: str
```